app/src/main/java/app/pachli/ViewMediaActivity.kt

```diff
@@ -213,10 +191,8 @@ class ViewMediaActivity : BaseActivity(), MediaActionsListener {
     }
 
     override fun onMediaTap() {
-        isToolbarVisible = !isToolbarVisible
-        for (listener in toolbarListeners) {
-            listener.onVisibilityChange(isToolbarVisible)
-        }
+        val isToolbarVisible = !viewModel.isToolbarVisible
+        viewModel.onToolbarVisibilityChange(isToolbarVisible)
```

Comment:
There's a (possibly theoretical, but good to clean up) race condition here.

I think the fix would be to adjust how the visibility is adjusted in the viewmodel.

Instead of `viewModel.onToolbarVisibilityChange` with a boolean parameter, implement it as a function that toggles the existing value. There's an atomic `getAndUpdate` function you can use to do this.

So replace 

```kotlin
    fun onToolbarVisibilityChange(isVisible: Boolean) {
        _toolbarVisibility.value = isVisible
    }
```

with 

```kotlin
    /**
     * Toggle the current state of the toolbar's visibility.
     *
     * @return The previous visiblity
     */
    fun toggleToolbarVisibility() = _toolbarVisibility.getAndUpdate { !it }
```

Then here on `onMediaTap()` you can replace these two lines with:

```kotlin
val isToolbarVisible = !toggleToolbarVisibility()
```

Reply:
Good point!

What do you think of the toggle method returning the new value? Returning the previous one sounded a bit strange to me.